src/sntp.rs

```diff
@@ -152,7 +158,12 @@ impl EspSntp {
         }
 
         unsafe {
-            sntp_set_time_sync_notification_cb(Some(Self::sync_cb));
+            if let Some(provided_cb) = callback {
```

Comment:
I think the user callback should work *in addition to* `Self::sync_cb`, not instead of `Self::sync_cb`.

Reply:
I don't think that's possible https://github.com/espressif/esp-idf/blob/6cfa88ed49b7d1209732347dae55578f4a679c98/components/lwip/apps/sntp/sntp.c#L64-L67